[Ong Han Wei] iP

rexong opened the pull request:
### DukeBot

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://www.google.com/)

DukeBot is your personal task bot that helps you remember your task! It's
- Text-based
- CLI
- ~~easy~~ Very Easy to use

All you need to do is 

1. download from the link [here](https://github.com/rexong/ip/releases/tag/A-Jar)
2. Double click duke.jar
3. Run the file
4. Add your task

It's **Free**! :smiley:

Features:

- [x] Managing tasks
- [ ] Managing Deadline (Coming soon) 
- [ ] Reminders (Coming soon)

Here is a random code in Java,
``` java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello world!");
    }
}
```
_Here is a random inline code._
`
System.out.println("Duke");
`

rexong committed:
Implemented Duke Level 1

rexong committed:
changed printMessage from public to private

rexong committed:
Added Constructor for Duke

rexong committed:
removed constructor, replace with attributs

rexong committed:
Implemented addTask and addTaskMessage

rexong committed:
Implemented printAllTasks

rexong committed:
refactor level 1 and level 2 and finish implementation of level 2

rexong committed:
refactor Duke, deleted TaskHandler

rexong committed:
copied Task.java from site

rexong committed:
Implemented Level-3

rexong committed:
Bugfix for regular task added

rexong committed:
Updated Task toString method

rexong committed:
implemented Level-4

rexong committed:
Implemented A-TextUiTesting

rexong committed:
Implemented Level-6

rexong committed:
Implemented Level-5

rexong committed:
Merge branch 'master' of https://github.com/rexong/ip

rexong committed:
Implemented Level-6

rexong committed:
Implemented Level 1

rexong committed:
implemented Level 2

rexong committed:
implement Level-3

rexong committed:
implemented Level-4

rexong committed:
implemented Level-5

rexong committed:
implemented Level-6

rexong committed:
Implement Storage class

rexong committed:
implement Level-7

rexong committed:
implemented Level-8

rexong committed:
Merge branch 'branch-Level-8'

rexong committed:
Implement A-MoreOOP

rexong committed:
Add duke.txt to gitignore file

rexong committed:
Implement A-Packages

rexong committed:
Implement A-Gradle

rexong committed:
implement A-Junit

rexong committed:
change error message display on console

rexong committed:
Reconfigure file structure

rexong committed:
Change error message display

rexong committed:
Implement A-JavaDoc

rexong committed:
Implement A-CodingStandard

rexong committed:
Implement Level-9

rexong committed:
Resolve merge conflict between master and branch-A-CodingStandard

rexong committed:
Resolve merge conflicts between master and branch-Level-9

rexong committed:
Fix find command to expect for a description not found

Jonaspng commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,207 @@
+    }
+
+    private static String getCommandWord(String command) {
+        if (command.startsWith("list")) {
```
Could consider switching to switch and case statements?

rexong replied:
was considering switch statements initially, however, as I'm using the method `startsWith`, I couldn't implement switch statement as the case for switch statement requires a string but the method `startsWith` return a boolean. If I want to implement switch statement, then I probably have to filter out the command word from the argument `command`.

Jonaspng replied:
Wouldn't it be worth the trouble?

Jonaspng commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,203 @@
+                String[] descriptions;
+                switch (taskType) {
+                case "[T]":
+                    description = texts[i].substring(10);
```
Maybe you could consider avoiding the magic number 10. You could consider using a named constant instead.
<img width="473" alt="image" src="https://user-images.githubusercontent.com/60882425/187228874-ae75ceda-47fa-42dc-ba82-1d9764ad9a53.png">

rexong replied:
Ah right, thanks for pointing this out!

Jonaspng commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,69 @@
+    private final String DUKE_INTRODUCTION = "Hello, I'm your personal assistant, Duke.";
+    private final String DUKE_HELP = "How can I assist you today?";
+    private final String DUKE_END = "Pleasure to be at your service! Run me again if you need more assistance! :)";
+    private final String divider = "__________________________________________________________";
```
Shouldn't this be named like a constant?

rexong replied:
yup, I think I missed this. Thanks!

rexong committed:
Add Checkstyle to gradle

Jonaspng reviewed (commented):
Looks good to me. Just some minor issues to fix.

rexong committed:
Update styles to comply with checkstyle

Jiacheng-y commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,92 @@
+            throw new TaskMarkException();
+        }
+        isDone = true;
+    }
```
Since this is a setter method for a Boolean variable, perhaps you can consider naming it `markDone()`

Jiacheng-y commented on src/main/java/duke/util/DateTimeHandler.java:
```diff
@@ -0,0 +1,45 @@
+    public static LocalDateTime formatDukeDateTime(String strDateTime) throws UnexpectedDateTimeFormatException {
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(strDateTime, dukeDateTimeFormat);
+            return dateTime;
```
I really like that you've split up the DateTimeFormatter variables out of the parse function! The code looks really neat! :)

Jiacheng-y commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,207 @@
+                : "";
+    }
+
+    private static int convertStringToInt(String number) throws IntegerExpectedException{
```
perhaps there should be a white space before the open braces : 

```java
private static int convertStringToInt(String number) throws IntegerExpectedException {
```

Jiacheng-y commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * Shows whether file is present or not.
+     * @return True if file is present else false.
```
perhaps `true` should be in lowercase since that's how the boolean `true` is represented in java

Jiacheng-y commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,94 @@
+     * @param ui Ui object to display message from storage
+     * @param obj Object to be written into the file.
+     */
+    public void update(Object obj, Ui ui) {
```
it may not be a wise choice for a public method to have a parameter with the most general class Object. I notice that it doesn't break your code or anything since every object has a `toString()` method but this allows anyone importing the Storage class to write anything into the file. I'm not sure that's a good idea.

Jiacheng-y reviewed (commented):
Generally looks good to merge! There are just a few nits to fix. The code is really well segregated and the classes and functions are quite comprehensive. I've learnt a lot from reading your code! :)

rexong committed:
Implement Level-10

rexong committed:
Update style to comply with checkstyle

rexong committed:
Merge branch 'branch-Level-10'

rexong committed:
Gradle run shadow on duke.Main

Duke starts on duke.Launcher instead of duke.Main due to JavaFX.

Let's update build.gradle to read from duke.Launcher.

Gradle can build Duke from duke.Launcher instead of duke.Main

rexong committed:
TaskList object can be created when the empty string is pass as parameters.

If empty string is passed in, TaskList object can be created without passing in an empty string.

Add an assertion line to check if the string passed in is empty.

Ensure that unnecessary parameters is used to create object.

rexong committed:
Merge pull request #1 from rexong/A-Assertions

TaskList object can be created when the empty string is pass as param…

rexong committed:
Refactor TaskCommand with a more meaningful name AddCommand

rexong committed:
Remove magic text from exceptions files

rexong committed:
Remove magic text from tasks files

rexong committed:
Remove magic literals from utils files

rexong committed:
Remove magic literals from duke files

rexong committed:
Refactor TaskList Constructor to have another method to ensure SLAP

rexong committed:
Merge branch 'master' into A-CodeQuality

rexong committed:
Merge pull request #2 from rexong/A-CodeQuality

A code quality

rexong committed:
Implement No Duplicated Task

rexong committed:
Merge pull request #3 from rexong/BCD-Extension

Implement No Duplicated Task

rexong committed:
Change background colour to pink when error message is shown, Change icon to Circle view

rexong committed:
Make window not resizable

rexong committed:
Merge pull request #5 from rexong/A-BetterGui

A better gui

rexong committed:
Change icon picture

rexong committed:
Change command messages to add personality

rexong committed:
Merge pull request #6 from rexong/A-Personality

A personality

rexong committed:
Fix checkStyle issues

rexong committed:
Add title to Window

rexong committed:
Update Readme.md

rexong committed:
Merge pull request #7 from rexong/A-UserGuide

Update Readme.md

rexong committed:
Change duke to shelbot

rexong committed:
Update Duke.java

rexong committed:
Update README.md

rexong committed:
Update README.md
